netcore/NetCore.Session/NetCoreSessionState.cs

```diff
@@ -0,0 +1,377 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using Apache.Geode.Client;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Apache.Geode.Session
+{
+    public class GeodeSessionStateValue
+    {
+        DateTime _lastAccessTimeUtc;
+        DateTime _expirationTimeUtc = DateTime.MinValue;
+        TimeSpan _spanUntilStale = TimeSpan.Zero;
+        private byte[] _value;
+
+        public GeodeSessionStateValue() { }
+        public GeodeSessionStateValue(byte[] value)
+        {
+            FromByteArray(value);
+        }
+
+        public byte[] Value
+        {
+            get { return _value; }
+            set { _value = value; }
+        }
+        public DateTime LastAccessTimeUtc
+        {
+            get { return _lastAccessTimeUtc; }
+            set { _lastAccessTimeUtc = value; }
+        }
+
+        public DateTime ExpirationTimeUtc
+        {
+            get { return _expirationTimeUtc; }
+            set { _expirationTimeUtc = value; }
+        }
+
+        public TimeSpan SpanUntilStale
+        {
+            get { return _spanUntilStale; }
+            set { _spanUntilStale = value; }
+        }
+
+        public byte[] ToByteArray()
+        {
+            int neededBytes = 3 * sizeof(long) + _value.Length;
+            byte[] byteArray = new byte[neededBytes];
+            int byteIndex = 0;
+
+            // Append LastAccessTimeUtc
```

Comment:
These comments do provide any meaningful information over the source below it. If the intent is to show some form of segmentation in the serialization then perhaps making them full blown methods.
```C#
byteIndex += AppendLastAccessTimeUtc(byteArray, byteIndex);
byteIndex += AppendExpirationTimeUtc(byteArray, byteIndex);
...

int AppendLastAccessTimeUtc(byte[] bytes, int offset) {
  Array.Copy(BitConverter.GetBytes(LastAccessTimeUtc.Ticks), 0, bytes, offset, sizeof(long));
  return sizeof(long);
}
```